src/utils/auth.js

```diff
@@ -0,0 +1,63 @@
+import { BASE_URL } from "./constants";
+
+function processResponse(res) {
+  if (res.ok) {
+    return res.json();
+  } else {
+    return Promise.reject(`An error just occurred: ${res.status}`);
+  }
+}
+
+export function register(email, password, name) {
+  return fetch(`${BASE_URL}/signup`, {
+    method: "POST",
+    headers: {
+      Accept: "application/json",
+      "Content-Type": "application/json",
+    },
+    body: JSON.stringify({ email, password, name }),
+  })
+    .then((res) => {
+      return processResponse(res);
+    })
+    .then((res) => {
+      return res;
+    });
+}
+
+export function login(email, password) {
+  return fetch(`${BASE_URL}/signin`, {
+    method: "POST",
+    headers: {
+      Accept: "application/json",
+      "Content-Type": "application/json",
+    },
+    body: JSON.stringify({ email, password }),
+  })
+    .then((res) => {
+      return processResponse(res);
+    })
+    .then((data) => {
+      if (data.token) {
+        localStorage.setItem("jwt", data.token);
+        return data;
+      } else {
+        return;
+      }
+    });
+}
+
+export function checkToken(token) {
+  return fetch(`${BASE_URL}/users/me`, {
+    method: "GET",
+    headers: {
+      Accept: "application/json",
+      "Content-Type": "application/json",
+      Authorization: `Bearer ${token}`,
+    },
+  })
+    .then((res) => {
+      return processResponse(res);
+    })
+    .then((data) => data);
```

Comment:
`    .then((data) => data);
`

you don't need this `then` here because you don't change the data